iot_verifier/src/hex_density.rs

```diff
@@ -22,29 +23,47 @@ impl HexResConfig {
     }
 }
 
-type HexMap = HashMap<H3Cell, u64>;
+type HexMap = HashMap<CellIndex, u64>;
 
-const DENSITY_TGT_RES: u8 = 4;
-const MAX_RES: u8 = 11;
-const USED_RES: Range<u8> = DENSITY_TGT_RES..MAX_RES;
-const SCALING_RES: Range<u8> = DENSITY_TGT_RES..MAX_RES + 2;
-
-static HIP17_RES_CONFIG: [Option<HexResConfig>; 11] = [
-    // Hex resolutions 0 - 3 and 11 and 12 are currently ignored when calculating density;
-    // For completeness sake their on-chain settings are N=2, TGT=100_000, MAX=100_000
-    None,                                 // 0
-    None,                                 // 1
-    None,                                 // 2
-    None,                                 // 3
-    Some(HexResConfig::new(1, 250, 800)), // 4
-    Some(HexResConfig::new(1, 100, 400)), // 5
-    Some(HexResConfig::new(1, 25, 100)),  // 6
-    Some(HexResConfig::new(2, 5, 20)),    // 7
-    Some(HexResConfig::new(2, 1, 4)),     // 8
-    Some(HexResConfig::new(2, 1, 2)),     // 9
-    Some(HexResConfig::new(2, 1, 1)),     // 10
+const MAX_RES: Resolution = Resolution::Eleven;
+const USED_RES: [Resolution; 7] = [
+    Resolution::Ten,
```

Comment:
huh, they went all out with the strong typing. Kinda annoying for this

Reply:
thankfully they attached a few nice features to the resolution enum but agreed; this particular use case made things a bit clumsier on our end. open to suggestions for adapting this back to a more "index-able" lookup the way it used to be indexing into a slice